pkgs/development/tools/build-managers/meson/setup-hook.sh

```diff
@@ -28,9 +28,39 @@ mesonConfigurePhase() {
         echo "meson: enabled parallel building"
     fi
 
+    if [[ ${checkPhase-ninjaCheckPhase} = ninjaCheckPhase && -z $dontUseMesonCheck ]]; then
+        checkPhase=mesonCheckPhase
+    fi
```

Comment:
Having this in `configurePhase` will only work when it is not overridden. But I probably would not bother with this too much until someone comes up with an use case.

Reply:
Yeah I think that's fine.

Reply:
I found only one usage of an overridden configurePhase with meson.
https://github.com/NixOS/nixpkgs/pull/259835